cmd/kubernetesDeploy.go

```diff
@@ -51,21 +51,41 @@ func runHelmDeploy(config kubernetesDeployOptions, utils kubernetes.DeployUtils,
 	if err != nil {
 		log.Entry().WithError(err).Fatalf("Container registry url '%v' incorrect", config.ContainerRegistryURL)
 	}
-	//support either image or containerImageName and containerImageTag
-	containerImageName := ""
-	containerImageTag := ""
 
-	if len(config.Image) > 0 {
-		containerImageName, containerImageTag, err = splitFullImageName(config.Image)
-		if err != nil {
-			log.Entry().WithError(err).Fatalf("Container image '%v' incorrect", config.Image)
+	helmValues := helmValues{}
+
+	if len(config.ImageNames) > 0 {
```

Comment:
This will be `true` all the time won't it? Like, `executeKaniko` and `cnbBuild` both populate `container/imageNameTag` **and** `container/imageNameTags` and as long as users have only **one** image in there, we want to also add `image.repository` and `image.tag` for compatibility reasons.

Reply:
Let me split that comment into understandeable chunks...

Reply:
> executeKaniko will always populate container/imageNameTag, but container/imageNameTags only if multi image is enabled.

Unfortunately that's not true for the `cnbBuild`, we are currently always putting data into the `container/imageNameTags` and `container/imageNames`:
https://github.com/SAP/jenkins-library/blob/2af9d3aae5b964355fd19757a44fccce8cee1b1b/cmd/cnbBuild.go#L510-L514

Reply:
Yes, we should populate `image.repository` and `image.tag` if there is only a single image in the array.

Reply:
Just recently (this morning) a PR #3545 was merged, that makes `kanikoExecute` always populate `container/imageNameTag` and `container/imageNameTags`. So `cnbBuild` is now in line with `kanikoExecute`.

The problem with custom pipelines still exists.

Reply:
> Yes, we should populate `image.repository` and `image.tag` if there is only a single image in the array.

We do that already

Reply:
I might have just misread @pbusko's comment.